web/client/components/TOC/Toolbar.jsx

```diff
@@ -330,14 +341,26 @@ class Toolbar extends React.Component {
                         </OverlayTrigger>
                         : null}
                     {this.props.activateTool.activateSwipeOnLayer && (status === 'LAYER') &&
-                        <OverlayTrigger
-                            key="layerSwipe"
-                            placement="top"
-                            overlay={<Tooltip id="layer-tooltip-swipe"><Message msgId="toc.compareTool" /></Tooltip>}>
-                            <Button key="layer-swipe" bsStyle={this.props?.swipeSettings?.active ? "success" : "primary"} className="square-button-md" onClick={() => this.showSwipeSettings(status)}>
-                                <Glyphicon glyph="transfer" />
-                            </Button>
-                        </OverlayTrigger>}
+                        <SplitButtonT
+                            onClick={() => this.showSwipeTools(status)}
+                            bsStyle={this.props?.swipeSettings?.active ? "success" : "primary"}
+                            {...splitToolButtonConfig}>
+                            <MenuItem
+                                active={this.props.swipeSettings.mode === "swipe"}
+                                onClick={() => this.props.onToolsActions.onSetSwipeMode("swipe")}>
+                                <Glyphicon glyph="vert-dashed"/><Message msgId="toc.swipe" />
+                            </MenuItem>
+                            <MenuItem
+                                active={this.props.swipeSettings.mode === "spy"}
+                                onClick={() => this.props.onToolsActions.onSetSwipeMode("spy")}>
+                                <Glyphicon glyph="search" /><Message msgId="toc.spyGlass" />
+                            </MenuItem>
+                            <MenuItem
+                                onClick={() => this.showConfiguration(status)}>
+                                <Glyphicon glyph="cog" /><Message msgId="toc.configureTool" />
+                            </MenuItem>
+                        </SplitButtonT>
```

Comment:
Can you wrap all the logic here into a single "SwipeButton" component? this is useful for future extraction that will come soon.
showSwipeTools, showConfiguration, splitToolButtonConfig etc... should go there.